### PR TITLE
Implemented the logic that refreshes the access token when a new memb…

### PR DESCRIPTION
**Pull request by ErkoRisthein**
…er joins

### CODE REVIEW

File: src/login/reducer.js [hunk 1]
```diff
@@ -67,13 +82,12 @@ export default function loginReducer(state = defaultState, action) {
       };
 
     case MOBILE_AUTHENTICATION_SUCCESS:
-      if (window.localStorage) {
-        localStorage.setItem(TOKEN_STORAGE_KEY, action.token);
-        localStorage.setItem(LOGIN_METHOD_STORAGE_KEY, 'mobileId');
-      }
+      updateLocalStorage(action, 'mobileId');
```
Comment (JordanValdma) [inline src/login/reducer.js]: normaalne 👍

File: src/login/reducer.js [hunk 2]
```diff
@@ -104,18 +118,34 @@ export default function loginReducer(state = defaultState, action) {
       return { ...state, loadingAuthentication: true, error: null };
 
     case ID_CARD_AUTHENTICATION_SUCCESS:
-      if (window.localStorage) {
-        localStorage.setItem(TOKEN_STORAGE_KEY, action.token);
-        localStorage.setItem(LOGIN_METHOD_STORAGE_KEY, 'idCard');
-      }
+      updateLocalStorage(action, 'idCard');
```
Comment (JordanValdma) [inline src/login/reducer.js]: normaalne 👍

File: src/login/reducer.js [hunk 3]
```diff
@@ -104,18 +118,34 @@ export default function loginReducer(state = defaultState, action) {
...
+        token: action.tokens.accessToken,
+        refreshToken: action.tokens.refreshToken,
         loadingAuthentication: false,
         loginMethod: 'idCard',
         error: null,
       };
 
+    case TOKEN_REFRESH_SUCCESS:
+      updateLocalStorage(action);
+      return { // reset all state so page is clean when entered again.
```
Comment (JordanValdma) [inline src/login/reducer.js]: unnecessary comment?

File: src/login/reducer.spec.js [hunk 4]
```diff
@@ -180,4 +189,26 @@ describe('Login reducer', () => {
     expect(newState.userConversionError).toBe('oh no!');
     expect(newState.loadingUserConversion).toBe(false);
   });
+
+  it('refreshes tokens in the state', () => {
```
Comment (JordanValdma) [inline src/login/reducer.spec.js]: maybe 'can refresh token'?

Comment (ErkoRisthein) [inline src/login/reducer.spec.js]: 🍺

**Pull request merged**